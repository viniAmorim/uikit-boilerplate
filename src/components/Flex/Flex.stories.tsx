import React from 'react';

import { FlexProps } from '@mantine/core';
import { Story, Meta } from '@storybook/react';

import Button from '../Button';
import Flex from './Flex';

export default {
  title: 'Atoms/Display/Flex',
  component: Flex,
} as Meta;

const Template: Story<FlexProps> = (args) => <Flex {...args} />;

export const Default = Template.bind({});
Default.args = {
  bg: 'rgba(0, 0, 0, .3)',
  mih: 50,
  children: (
    <>
      <Button variant="secondary">Button 1</Button>
      <Button variant="secondary">Button 2</Button>
      <Button variant="secondary">Button 3</Button>
    </>
  ),
};

export const Column = Template.bind({});
Column.args = {
  bg: 'rgba(0, 0, 0, .3)',
  mih: 50,
  children: (
    <>
      <Button variant="secondary">Button 1</Button>
      <Button variant="secondary">Button 2</Button>
      <Button variant="secondary">Button 3</Button>
    </>
  ),
  direction: 'column',
};

export const Center = Template.bind({});
Center.args = {
  bg: 'rgba(0, 0, 0, .3)',
  mih: 50,
  children: (
    <>
      <Button variant="secondary">Button 1</Button>
      <Button variant="secondary">Button 2</Button>
      <Button variant="secondary">Button 3</Button>
    </>
  ),
  align: 'center',
  justify: 'center',
};
