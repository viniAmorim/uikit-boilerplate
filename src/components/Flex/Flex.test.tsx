import React from 'react';

import { render } from '@testing-library/react';

import Flex from './Flex';

describe('Flex', () => {
  it('should render correctly', () => {
    const { container } = render(<Flex />);

    const flex = container.firstChild;

    expect(flex).toBeInTheDocument();
    expect(flex).toMatchSnapshot();
  });
  it('should render children with correct flexbox properties', () => {
    const { container } = render(
      <Flex justify="center" align="center">
        <div>Child 1</div>
        <div>Child 2</div>
      </Flex>,
    );
    const flex = container.firstChild;
    expect(flex).toHaveStyle(`
      display: flex;
      justify-content: center;
      align-items: center;
    `);
  });
});
