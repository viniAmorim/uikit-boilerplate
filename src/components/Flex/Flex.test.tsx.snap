// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex should render correctly 1`] = `
<div
  class="mantine-1jggmkl"
/>
`;
